dispatch/dispatch.go

```diff
@@ -31,20 +31,79 @@ func Split(chIn interface{}, chsOut ...interface{}) {
 			return
 		}
 		for _, chOut := range chsOut {
-			switch reflect.TypeOf(chOut).Kind() {
-			case reflect.Array, reflect.Slice:
-				for i := 0; i < reflect.ValueOf(chOut).Len(); i++ {
-					if reflect.ValueOf(chOut).Index(i).Kind() != reflect.Chan {
-						panic(fmt.Sprintf("cannot split to value of type %T", chOut))
-					}
-					reflect.ValueOf(chOut).Index(i).Send(msg)
-				}
-			case reflect.Chan:
-				reflect.ValueOf(chOut).Send(msg)
-			default:
+			SendToInterface(chOut, msg)
+		}
+	}
+}
+
+func SendToInterface(chOut interface{}, msgValue reflect.Value) {
```

Comment:
The `ToInterface` is overly explicit given the type definition of the function, the name should be `Send (ch interface{}, val reflect.Value)`.